packages/react-components/react-toast/src/components/Timer.tsx

```diff
@@ -0,0 +1,47 @@
+import * as React from 'react';
+import { makeStyles } from '@griffel/react';
+
+const useStyles = makeStyles({
+  progress: {
+    animationName: {
+      from: {
+        opacity: 0,
+      },
+      to: {
+        opacity: 0,
+      },
+    },
+  },
+});
+
+export const Timer: React.FC<{
+  running: boolean;
+  timeout: number;
+  onTimeout: () => void;
+}> = props => {
+  const styles = useStyles();
+  const { running, timeout, onTimeout } = props;
+  const cleanupRef = React.useRef<() => void>(() => null);
+
+  const bindEventListeners = React.useCallback(
+    (el: HTMLElement) => {
+      cleanupRef.current();
+      if (el) {
+        el.addEventListener('animationend', onTimeout);
+        cleanupRef.current = () => el.removeEventListener('animationend', onTimeout);
+      }
+    },
+    [onTimeout],
+  );
```

Comment:
Why this is not inside `useEffect`?

- an element will never change
- you can handle dependencies in `useEffect`, too

Reply:
at some point I had `animationstart` there and I found that it wasn't triggering the event listener if I put it useEffect. I don't use that event anymore so I can put it in an effect now